src/Tasks/Microsoft.NET.Build.Tasks/GetPackagesToPrune.cs

```diff
@@ -276,6 +251,66 @@ static Dictionary<string, NuGetVersion> LoadPackagesToPruneFromTargetingPack(Log
+            bool useFrameworkPackageData = !targetFrameworkIdentifier.Equals(".NETCoreApp") || targetVersion.Major < PrunePackageDataMinMajorVersion;
+
+            Dictionary<string, NuGetVersion> packages = null;
+
+            if (useFrameworkPackageData)
+            {
+                packages = LoadPackagesToPruneFromFrameworkPackages(targetFrameworkIdentifier, targetFrameworkVersion, frameworkReference);
+                if (packages != null)
+                {
+                    log.LogMessage("Loaded prune package data from framework packages");
+                }
+                else
+                {
+                    log.LogMessage("Failed to load prune package data from framework packages");
+                }
+            }
+            else
+            {
+                log.LogMessage("Loading prune package data from PrunePackageData folder");
+                packages = LoadPackagesToPruneFromPrunePackageData(targetFrameworkIdentifier, targetFrameworkVersion, frameworkReference, prunePackageDataRoot);
+
+                //  For the version of the runtime that matches the current SDK version, we don't include the prune package data in the PrunePackageData folder.  Rather,
+                //  we can load it from the targeting packs that are packaged with the SDK.
+                if (packages == null)
+                {
+                    log.LogMessage("Failed to load prune package data from PrunePackageData folder, loading from targeting packs instead");
+                    packages = LoadPackagesToPruneFromTargetingPack(log, targetFrameworkIdentifier, targetFrameworkVersion, frameworkReference, targetingPackRoots);
+                }
+
+                //  Fall back to framework packages data for older framework for WindowsDesktop if necessary
+                //  https://github.com/dotnet/windowsdesktop/issues/4904
+                if (packages == null && frameworkReference.Equals("Microsoft.WindowsDesktop.App", StringComparison.OrdinalIgnoreCase))
+                {
+                    log.LogMessage("Failed to load prune package data for WindowsDesktop from targeting packs, loading from framework packages instead");
+                    packages = LoadPackagesToPruneFromFrameworkPackages(targetFrameworkIdentifier, targetFrameworkVersion, frameworkReference,
+                        acceptNearestMatch: true);
+                }
+            }
+
+            //  If LoadPrunePackageDataFromNearestFramework is true and we still haven't found data, try the previous framework version
+            if (packages == null && loadPrunePackageDataFromNearestFramework)
+            {
+                // If we can go to a lower version, recursively try it
+                if (targetVersion.Major > MinSupportedFrameworkMajorVersion)
+                {
+                    string fallbackVersion = $"{targetVersion.Major - 1}.0";
```

Comment:
What if we have frameworks with minor versions?

Reply:
I think it's unlikely we'll ship minor versions again.  I'm OK with this breaking or not working correctly if we eventually do create a minor version.